kornia/color/hls.py

```diff
@@ -65,19 +63,17 @@ def rgb_to_hls(image: Tensor, eps: float = 1e-8) -> Tensor:
     l_ /= 2  # luminance
 
     # note that r,g and b were previously div by (max - min)
-    r = torch.select(im, -3, 0)
-    g = torch.select(im, -3, 1)
-    b = torch.select(im, -3, 2)
+    r, g, b = im[..., 0, :, :], im[..., 1, :, :], im[..., 2, :, :]
     # h[imax == 0] = (((g - b) / (max - min)) % 6)[imax == 0]
     # h[imax == 1] = (((b - r) / (max - min)) + 2)[imax == 1]
     # h[imax == 2] = (((r - g) / (max - min)) + 4)[imax == 2]
     cond = imax.unsqueeze(-3) == _RGB2HSL_IDX
     if image.requires_grad:
-        h = torch.mul((g - b) % 6, torch.select(cond, -3, 0))
+        h = torch.mul((g - b) % 6, cond[..., 0, :, :])
     else:
-        torch.mul((g - b).remainder(6), torch.select(cond, -3, 0), out=h)
-    h += torch.add(b - r, 2) * torch.select(cond, -3, 1)
-    h += torch.add(r - g, 4) * torch.select(cond, -3, 2)
+        torch.mul((g - b).remainder(6), cond[..., 0, :, :], out=h)
+    h += torch.add(b - r, 2) * cond[..., 1, :, :]
+    h += torch.add(r - g, 4) * cond[..., 2, :, :]
```

Comment:
isn't better without torch add, mul, etc? cc @ducha-aiki @edgarriba 
```suggestion
    h = h + (b - r + 2) * cond[..., 1, :, :]
    h = h + (r - g + 4) * cond[..., 2, :, :]
```

Reply:
+1